### ISSUE TITLE
File path and content markers

### ISSUE THREAD

**Issue by gomoto** [gomoto/html-injector#6]
Right now, xxx marks file path, and yyy marks file content.
Is there a better way?

Idea:

```
<script src="{filePathTransform}">
  {{fileContentTransform}}
</script>
```

Where would these transforms get defined?

**Comment by gomoto** [comment]
Define transforms as regular expressions?

```
html-injector --transform-path '/src/build/g' --transform-content '/secret/xxx/g' index.html templates src/**/*.html
```

```
injector('index.html', 'templates', 'src/**/*.html', {
  transformPath: '/src/build/g',
  transformContent: '/secret/xxx/g'
});
```

**Comment by gomoto** [comment]
What about named transforms? Define them in package.json?

```
{
  "html-injector": {
    "convertSrcToBuild": "/src/build/g",
    "hideSecrets": "/secret/xxx/g"
  }
}
```

Transform names map directly to templates. Don't need to use flags. 

```
<script src="{convertSrcToBuild}">
  {{hideSecrets}}
</script>
```

Single brackets mark file paths. Double brackets mark file content.

**Comment by gomoto** [comment]
[Regular expressions in JSON may cause problems](https://github.com/kangax/html-minifier/blob/18c486d25369599bab9381ba9d45bbf3c2107924/cli.js#L46)

Use .js file? Define all named transforms as functions in htmlInjector.js, and pass the original value as a parameter. Also pass the type (path or content). Functions should return value that will replace original.

```
// htmlInjector.js
module.exports = {
  convertSrcToBuild: (original, type) => {},
  hideSecrets: (original, type) => {}
}
```

**Comment by gomoto** [comment]
Multiple transforms?

```
<script src="{ convertSrcToBuild mangleName }">
  {{ hideSecrets mangleName }}
</script>
```

**Comment by gomoto** [comment]
No transforms?

```
<script src="{}">
  {{}}
</script>
```

Looks weird.

**Comment by gomoto** [comment]
Instead of single and double brackets, provide 'path' and 'content' built-in transforms.

```
<script src="{ path }">
  { content }
</script>
```

These are not really transforms; they just return path or content, no matter what the input.

**Comment by gomoto** [comment]
By default, html-injector searches for htmlInjector.js in root directory.
Provide a flag to specify a different file path.

```
html-injector --transforms /path/to/file index.html js src/**/*.js
```

**Comment by gomoto** [comment]
The multiple transforms example from above becomes:

```
<script src="{ path convertSrcToBuild mangleName }">
  { content hideSecrets mangleName }
</script>
```

**Comment by gomoto** [comment]
Brackets + transforms syntax implemented in #7
